### ISSUE TITLE
function with for...in or for...of removed during tree-shaking

### ISSUE THREAD

**Issue by Zirro** [rollup/rollup#870]
I've extracted a minimal testcase that reproduces the issue in Rollup 0.34.9:

```javascript
const items = document.querySelector('.items');

function sayHello() {
  for (const item of items.children) {
    item.textContent = 'Hello!';
  }
}

sayHello();
```

The function remains after tree-shaking if I remove the loop but keep the .textContent-line, or add more code such as a console.log() into it.

**Comment by bfred-it** [comment]
It looks like it has trouble with the variable name item.

[This works](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%2C%22moduleId%22%3A%22%22%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22const%20items%20%3D%20document.querySelector('.items')%3B%5Cn%5Cnfunction%20sayHello()%20%7B%5Cn%20%20for%20(const%20child%20of%20items.children)%20%7B%5Cn%20%20%20%20item.textContent%20%3D%20'Hello!'%3B%5Cn%20%20%7D%5Cn%7D%5Cn%5CnsayHello()%3B%22%7D%5D%7D) (just different var name): 


```js
const items = document.querySelector('.items');

function sayHello() {
  for (const child of items.children) {
    item.textContent = 'Hello!';
  }
}

sayHello();
```

But [this doesn't:](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%2C%22moduleId%22%3A%22%22%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22const%20items%20%3D%20document.querySelector('.items')%3B%5Cn%5Cnfunction%20sayHello()%20%7B%5Cn%20%20for%20(var%20item%20%3D%200%3B%3B)%20%7B%5Cn%20%20%20%20item.textContent%20%3D%20'Hello!'%3B%5Cn%20%20%7D%5Cn%7D%5CnsayHello()%3B%22%7D%5D%7D)

```js
const items = document.querySelector('.items');

function sayHello() {
  for (var item = 0;;) {
    item.textContent = 'Hello!';
  }
}
sayHello();
```

**Comment by Zirro** [comment]
Having looked at issue #786, I believe the cause might be the same. Feel free to close this issue as a dupe if you can verify that this is the case.

**Comment by Rich-Harris** [comment]
Sorry for the delay – fixed in 0.35.4
